bw_add_sshkeys.py

```diff
@@ -220,36 +166,29 @@ def ssh_add(session, item_id, key_id):
     logging.debug('Item ID: %s', item_id)
     logging.debug('Key ID: %s', key_id)
 
-    # FIXME: avoid temporary files, if possible (StringIO ?)
-    with tempfile.NamedTemporaryFile() as tmpfile:
-        proc = subprocess.Popen(
-            list(filter(None, [
-                'bw',
-                (None, '--nointeraction')[cli_supports('nointeraction')],
-                '--quiet',
-                'get',
-                'attachment', key_id,
-                '--itemid', item_id,
-                '--output', tmpfile.name,
-                '--session', session
-            ])),
-            stderr=subprocess.PIPE
-        )
-        (_, stderr) = proc.communicate()
-        if proc.returncode:
-            logging.error(stderr.decode('utf-8'))
-            return False
-
-        logging.debug("Running ssh-add")
-
-        # CAVEAT: `ssh-add` provides no useful output, even with maximum verbosity
-        proc = subprocess.Popen(['ssh-add', tmpfile.name])
-        proc.wait()
+    proc_attachment = subprocess.run([
+            'bw',
+            'get',
+            'attachment', key_id,
+            '--itemid', item_id,
+            '--raw',
+            '--session', session
+        ],
+        stdout=subprocess.PIPE,
+        text=True,
+        check=True,
+    )
+    ssh_key = proc_attachment.stdout
 
-        if proc.returncode:
-            return False
+    logging.debug("Running ssh-add")
 
-        return True
+    # CAVEAT: `ssh-add` provides no useful output, even with maximum verbosity
+    subprocess.run(
+        ['ssh-add', '-'],
+        input=ssh_key,
```

Comment:
Hum, that's unfortunate...

Reply:
Fixed in last commit. When adding the `SSH_ASKPASS_REQUIRE=never` env variable, `ssh-add` gets the passphrase from the terminal. Not exactly sure how it works since `stdin` is redirected, but it works.

The downside is that it will never use the graphical `askpass` program, like `x11-ssh-askpass`. I expected the `prefer` value to fallback on the terminal if `askpass` is absent, but it fails with the same error...

Reply:
The thing is you can have a different askpass like https://archlinux.org/packages/community/x86_64/openssh-askpass/.

Honestly I think this is fine as is, you need to run the script in a terminal to unlock bitwarden anyway.